src/Http/Http.Abstractions/src/Routing/EndpointMetadataCollection.cs

```diff
@@ -156,6 +157,24 @@ public EndpointMetadataCollection(params object[] items)
     /// <returns>An <see cref="IEnumerator"/> of all metadata items.</returns>
     IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
 
+    /// <inheritdoc/>
+    public override string ToString()
```

Comment:
I don't think we should override this collection ToString. If someone wants to view the contents then they can do it with a debugger.

Instead, put a debugger display on the collection:

```cs
[DebuggerDisplay("Count = {Count}")]
```

If you want, you could also add a debugger proxy. `List<T>` does this for example: https://github.com/dotnet/runtime/blob/73818601becad0d1c6cfc58b5d947db010da2d6b/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs#L17-L18. A debugger proxy changes what properties are visible in IDEs when debugging.

You'll need to copy and paste this internal type: https://github.com/dotnet/runtime/blob/73818601becad0d1c6cfc58b5d947db010da2d6b/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ICollectionDebugView.cs
Put it under `src/Http/Http.Abstractions/src/Internal` and update the namespace to match other files in that directory.